salt/states/x509.py

```diff
@@ -629,6 +629,9 @@ def pem_managed(name,
         ret['comment'] = 'The file {0} will be updated.'.format(name)
         return ret
 
+    ret['changes']['new'] = new
```

Comment:
If we have changes, we also want the changes dictionary in the state output when `test=True`. Can you move these lines to above the `if __opts__['test'] is True` check?